lib/dal.js

```diff
@@ -0,0 +1,86 @@
+var fs = require('fs');
+var Q = require('q');
+
+/** Reads from the configuration file and returns data
+ * @param {string} confFile - the path and filename to the configuration file
+ * @return {Q} - Q promise object
+ */
+function getConfig(confFile) {
+  var readJSON = Q.denodeify(fs.readFile.bind(fs.readFile));
+  return readJSON(confFile, 'utf8')
+    .then(function (data) {
+      var JSONdata = JSON.parse(data);
+      return Q.resolve(JSONdata);
+    });
+}
+
+/** Attempts to access configuration file data for the shared extension
+ * @param {string} confFile - the path and filename to the configuration file
+ * @param {string} name - the name of the sharedExtension to access
+ * @return {Q} - Q promise object
+ */
+
+var getSharedExtension = function(confFile, name) {
+  var defer = Q.defer();
+  var sharedExtension = {};
+  sharedExtension.pos = -1;
```

Comment:
You could also have a boolean called found to determine this since -1 is rather cryptic.